tests/openassetio-core-c/private/managerAPI/CManagerInterfaceTest.cpp

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+#include <catch2/catch.hpp>
+#include <catch2/trompeloeil.hpp>
+
+#include <openassetio/c/errors.h>
+#include <openassetio/c/namespace.h>
+// private headers
+#include <managerAPI/CManagerInterface.hpp>
+
+// Duplicated from CManagerInterface.
+constexpr size_t kStringBufferSize = 500;
+
+/**
+ * Mock manager API implementation that the function pointer suite (see
+ * `getSuite`) will delegate to.
+ */
+struct MockCAPI {
+  MAKE_MOCK1(dtor, void(OPENASSETIO_NS(managerAPI_ManagerInterface_h)));
+
+  MAKE_MOCK3(identifier, int(OPENASSETIO_NS(SimpleString) *, OPENASSETIO_NS(SimpleString) *,
+                             OPENASSETIO_NS(managerAPI_ManagerInterface_h)));
+
+  MAKE_MOCK3(displayName, int(OPENASSETIO_NS(SimpleString) *, OPENASSETIO_NS(SimpleString) *,
+                              OPENASSETIO_NS(managerAPI_ManagerInterface_h)));
+};
+
+/**
+ * Get a ManagerInterface C API function pointer suite that assumes the
+ * provided `handle` is a `MockCAPI` instance.
+ */
+OPENASSETIO_NS(managerAPI_ManagerInterface_s) getSuite() {
+  return {// dtor
+          [](OPENASSETIO_NS(managerAPI_ManagerInterface_h) h) {
+            auto *api = reinterpret_cast<MockCAPI *>(h);
+            api->dtor(h);
+          },
+          // identifier
+          [](OPENASSETIO_NS(SimpleString) * err, OPENASSETIO_NS(SimpleString) * out,
+             OPENASSETIO_NS(managerAPI_ManagerInterface_h) h) {
+            auto *api = reinterpret_cast<MockCAPI *>(h);
+            return api->identifier(err, out, h);
+          },
+          // displayName
+          [](OPENASSETIO_NS(SimpleString) * err, OPENASSETIO_NS(SimpleString) * out,
+             OPENASSETIO_NS(managerAPI_ManagerInterface_h) h) {
+            auto *api = reinterpret_cast<MockCAPI *>(h);
+            return api->displayName(err, out, h);
+          }};
+}
+
+SCENARIO("A CManagerInterface is destroyed") {
+  GIVEN("An opaque handle and function suite") {
+    MockCAPI capi;
+
+    auto *handle = reinterpret_cast<OPENASSETIO_NS(managerAPI_ManagerInterface_h)>(&capi);
+    auto const suite = getSuite();
+
+    THEN("CManagerInterface's destructor calls the suite's dtor") {
+      REQUIRE_CALL(capi, dtor(handle));
+
+      openassetio::managerAPI::CManagerInterface cManagerInterface{handle, suite};
+    }
+  }
+}
+
+SCENARIO("A host calls CManagerInterface::identifier") {
+  GIVEN("A CManagerInterface wrapping an opaque handle and function suite") {
+    MockCAPI capi;
+
+    auto *handle = reinterpret_cast<OPENASSETIO_NS(managerAPI_ManagerInterface_h)>(&capi);
+    auto const suite = getSuite();
+
+    // Expect the destructor to be called, i.e. when cManagerInterface
+    // goes out of scope.
+    // Mysteriously, this must come _before_ the construction
+    // of cManagerInterface...
+    REQUIRE_CALL(capi, dtor(handle));
```

Comment:
Ah, this is a "feature" of Trompeloeil (IIRC it's the same with Google Mock), in that any unexpected call is an error, so you must `REQUIRE` all of them.  

There is a softer alternative `ALLOW_CALL`, which takes the same arguments and, as it implies, allows the call, but doesn't make the call required, if that's preferable?

Maybe there's a way to construct the mock and its common expectations in a function to hide this, but I suspect it might be tricky, since the expectations are, by default at least, asserted at the end of the current scope.